client/client-multi/client-c/src/kaa/kaa_extension_private.h

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright 2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+/** @file
+ * This file defines all extensions for the current client
+ * instance. Currently, this file is static but will be auto-generated
+ * in the future.
+ *
+ * kaa_extension.c MUST use this file. That's a requirement, so we can
+ * easily replace this file with auto-generated one in the future.
+ *
+ * @warning this file is a private part of the kaa_extension.c and
+ * must only be used from that file. If you need extension instance,
+ * use interface defined in the kaa_extension.h. (That is the reason
+ * why the file doesn't have include guard.)
+ */
+#include <kaa_common.h>
+
+#include "kaa_private.h"
+
+// Extension is not really an extension. It doesn't have any context
+// and its implementation is embedded in platfrom_protocol.
+
+// They will be extern, thus prefixed.
+static struct kaa_extension kaa_extension_bootstrap = {
```

Comment:
To keep things simpler we can introduce macro that will initialize kaa_extension struct and use it  kaa_extension[] array directly